apps/meteor/app/livechat/server/api/v1/room.ts

```diff
@@ -59,14 +59,14 @@ API.v1.addRoute('livechat/room', {
 				return API.v1.success({ room, newRoom: false });
 			}
 
-			let agent;
+			let agent: SelectedAgent | undefined;
 			const agentObj = agentId && (await findAgent(agentId));
 			if (agentObj) {
 				if (isAgentWithInfo(agentObj)) {
 					const { username = undefined } = agentObj;
-					agent = { agentId, username };
+					agent = { agentId, username } as SelectedAgent;
```

Comment:
this type cast looks wrong.. `username` cannot be `undefined` in `SelectedAgent`..